### ISSUE TITLE
net/freeradius additional options

### ISSUE THREAD

**Issue by pgaultier** [opnsense/plugins#939]
Hi,

I have migrated from pfSense to OPNSense and I have a little remark about the way freeradius is handled.

In my pfSense configuration I was able to add the `Login-Time:` parameter to the users, and many others... 
As far as I can understand OPNSense it is not possible to use options which are not defined in the GUI.

Is it possible to add some field to the `User` configuration such as `AdditionalOptions` where we can add the wanted options without creating a really complex GUI which would be able to handle all of the freeradius options ?

This way, I would be able to add in this `AdditionalOptions` field specific parameters for each user : 
```
Login-Time := "Al0800-2100",
Simultaneous-Use := "1"
```

**Comment by pgaultier** [comment]
The two options, I use are : 
 * Login-Time
 * Simultaneous-Use

Thank you ;-)

**Comment by mimugmail** [comment]
Do you have some FreeRadius background? Sim Use requires SQL I'd guess, Login-Time too?

**Comment by pgaultier** [comment]
On my pfSense I don't use SQL and Simultaneous-Use is working (same for Login-Time)

**Comment by pgaultier** [comment]
One last thing, can you make the validation regex a little more relaxed, the chars `{` and `}` cannot be used through the interface (validation error while those chars are allowed by freeradius). I had to manually update the config.xml to allow passwords such as `{my_0wn_p4ssword}`.

As far as I can understand the opnsense system,

```
<mask>/^([0-9a-zA-Z._\-\!\$\%\/\(\)\+\#\=]){1,128}$/u</mask>
```

should be replaced with

```
<mask>/^([0-9a-zA-Z._\-\!\$\%\/\(\)\+\#\=\{\}]){1,128}$/u</mask>
```

**Comment by fabianfrz** [comment]
@mimugmail you may want to manage the attributes like the SNI upstream maps in #930 in the future. It would make your configuration more flexible to allow adding fields manually while still validating them in the backend.

**Comment by pgaultier** [comment]
That would be awesome because those parameters are really specifics and optional

**Comment by AdSchellevis** [comment]
I would be careful in converting all kinds of options into a key/value store in cases where you can't validate your data uniformly.  

`Login-Time` and `Simultaneous-Use` are very different types of data, which would likely result in quite some additional logic in the model and/or controller to handle them properly.

The `JsonKeyValueStoreField` was originally added to support options list transported using config, like Tinc receiving valid ciphers using:
https://github.com/opnsense/plugins/blob/4368fa910ada7f32dd4b9ac2f6494ccdaf53a06e/security/tinc/src/opnsense/mvc/app/models/OPNsense/Tinc/Tinc.xml#L58-L64

**Comment by pgaultier** [comment]
@AdSchellevis I totally agree with you, but letting the end user add "missing" options in "advanced" mode will make everything more configurable.
The main question would be to know which options should be considered as "advanced" and could be managed this way because in advanced mode the validation should be omitted.

**Comment by mimugmail** [comment]
Advanced Mode means it's only visible when you click on it. We still will make for every option a button or text field.

Within the last year there were not many requests so this is OK

**Comment by AdSchellevis** [comment]
@pgaultier Advanced isn't a good reason to ignore validations in my humble opinion, we specifically have chosen to not include text field options to generate configurations for this reason.

Quite some people think they now what they're doing, leading to "advanced users" reporting more and more self inflicted issues.

In some scenario's it's possible to use a custom template to generate your config, which isn't available in the UI (for example in more complex proxy scenario's), I'm not sure if there's an easy option to create custom overrides here (not really convinced it's needed either).

Thinking about better ways to present less common options is always a good idea, which doesn't hurt our philosophy and helps people finding their way in the system easier.

**Comment by pgaultier** [comment]
@AdSchellevis I understand and I partially agree with you. I agree with you about the validation of standard configuration which is really important to avoid errors but is it a reason to lock out users who would like to use some "advanced" or "not common" configurations ?

Anyway, adding the two fields and updating the users config file generation would perfectly fit my needs, but something a little more flexible would be great (and even if I have no example in mind, it would be probably a good idea for other modules).
